scantool/diag_l0_elm.c

```diff
@@ -752,6 +766,17 @@ elm_initbus(struct diag_l0_device *dl0d, struct diag_l1_initbus_args *in)
 					break;
 			}
 
+			// set wakeup message if applicable
+			if (dev->wm != NULL) {
+				uint8_t wm[18];
+				sprintf((char *) wm, "ATWM %02X %02X %02X %02X\x0D", dev->wm->data[0], dev->wm->data[1], dev->wm->data[2], dev->wm->data[3]);
+				rv=elm_sendcmd(dl0d, wm, 17, 500, NULL);
+				if (rv < 0) {
+					fprintf(stderr, FLFMT "elm_initbus: ATWM failed\n", FL);
+					return diag_iseterr(DIAG_ERR_GENERAL);
+				}
+			}
+
```

Comment:
these two identical chunks of code could probably split into a "elm_updatewm()" function ?